Source/WebKit/WebProcess/WebPage/Cocoa/TextAnimationController.mm

```diff
@@ -272,12 +272,18 @@
 
     auto replacedRangeAfterReplace = WebCore::resolveCharacterRange(*sessionRange, replacedCharacterRange, defaultTextAnimationControllerTextIteratorBehaviors);
 
-    auto unstyledRange = *sessionRange;
-    unstyledRange.start.container = replacedRangeAfterReplace.endContainer();
-    unstyledRange.start.offset = replacedRangeAfterReplace.endOffset();
+    auto endPosition = makeContainerOffsetPosition(sessionRange->start);
+    auto endBoundaryPoint = makeBoundaryPoint(endOfEditableContent(endPosition));
 
-    auto unstyledRangeUUID = WTF::UUID::createVersion4();
-    m_unstyledRange = { unstyledRangeUUID, unstyledRange };
+    WTF::UUID unanimatedRangeUUID { WTF::UUID::emptyValue };
+    if (endBoundaryPoint) {
+        auto unanimatedRange = makeSimpleRangeHelper(replacedRangeAfterReplace.end, *endBoundaryPoint);
+
+        if (unanimatedRange) {
+            unanimatedRangeUUID = WTF::UUID::createVersion4();
+            m_unanimatedRangeData = { unanimatedRangeUUID, *unanimatedRange };
+        }
+    }
```

Comment:
Should we be resetting `m_unanimatedRangeData` to `std:nullopt` in an else case here?

Reply:
also, did you make sure that the smart replies bug is still fixed with this change?